src/main/java/com/github/davidmoten/rtree/RTree.java

```diff
@@ -891,39 +860,29 @@ public Visualizer visualize(int width, int height) {
     }
 
     private Rectangle calculateMaxView(RTree<T, S> tree) {
-        return tree.entries().reduce(Optional.<Rectangle>absent(),
-                new Func2<Optional<Rectangle>, Entry<T, S>, Optional<Rectangle>>() {
-
-                    @Override
-                    public Optional<Rectangle> call(Optional<Rectangle> r, Entry<T, S> entry) {
-                        if (r.isPresent())
-                            return of(r.get().add(entry.geometry().mbr()));
-                        else
-                            return of(entry.geometry().mbr());
-                    }
-                }).toBlocking().single().or(rectangle(0, 0, 0, 0));
+        return tree.entries()
+                .reduce(Optional.empty(), rectangleAccumulator)
+                .toBlocking().single()
+                .orElse(ZERO_RECTANGLE);
     }
 
     public Optional<? extends Node<T, S>> root() {
         return root;
     }
 
     /**
-     * If the RTree has no entries returns {@link Optional#absent} otherwise returns
+     * If the RTree has no entries returns {@link Optional#empty()} otherwise returns
      * the minimum bounding rectangle of all entries in the RTree.
-     * 
+     *
      * @return minimum bounding rectangle of all entries in RTree
      */
     public Optional<Rectangle> mbr() {
-        if (!root.isPresent())
-            return absent();
-        else
-            return of(root.get().geometry().mbr());
+        return root.map(r -> r.geometry().mbr());
```

Comment:
As I remember you had a comment about this lambda which is disappeared somehow. Actually it can be constant if you worry about instantiating the lambda expression, but I think Optional.map is designed for this.